services/grpc/internal/transformations/graph_objects.go

```diff
@@ -0,0 +1,36 @@
+package transformations
+
+type Node struct {
+	NodeId string `db:"nodeId"`
+	Doc string `db:"doc"`
```

Comment:
I like the idea of a document being a scope and not fundamental to the node.

Reply:
Okay that makes sense, maybe I'll call it a `Diff` then? Or maybe we don't need it because it's stored on the edge?
One idea that would avoid needing to traverse paths would be to store the current state of the compiled document at each node

Reply:
Actually yea it seems like `Diff` is what is stored on each edge. On the node we either don't need a state, or we can compute and cache the entire document (trading storage for compute time). Duplicating data like this seems fine for small documents, but maybe we do a sort of checkpointing where we only do it in a subset of the nodes?

Reply:
> One idea that would avoid needing to traverse paths would be to store the current state of the compiled document at each node 

If we have some notion of chunking that could make sense. That prioritizes CPU over storage (i.e. we only need to calculate things once, and then we just store the chunk). I think in the beginning we should just recalculate everything and add a notion of chunking later.

Reply:
Lol, just read your second comment here. All agreed. 

Intuitively it feels like checkpointing should be conducted at branching points, but let's play around before implementing anything.